tests/framework/project/test_pipeline_registry.py

```diff
@@ -50,6 +46,11 @@ def register_pipelines():
     sys.path.pop(0)
 
 
+def test_pipelines_without_configure_project_throws_exception():
+    with pytest.raises(ModuleNotFoundError, match="No module named 'dummy_package'"):
+        assert pipelines == {}
+
+
```

Comment:
This seems potentially confusing as an error message, though I'm not sure when in reality it might actually occur? Really what's happened here is there hasn't been a call to `configure_project` and hence `pipelines` doesn't exist yet (I think), but a ModuleNotFoundError doesn't give any clue of that or how you might fix it.

What happens now/do we have tests for the following cases?
1. there's no `pipeline_registry.py` file at all, and I do some `kedro` command
2. `pipeline_registry.py` exists but has no `register_pipelines` function
I was kind of expecting to see these in this file already but don't think they're here 🤔 Maybe worth adding them or just manually testing. 

It looks like `_get_pipelines_registry_callable` should already raise a clear error message in these cases but worth double checking since this tripped us up before.